addon/components/frost-icon.js

```diff
@@ -1,22 +1,54 @@
-import Ember from 'ember'
-import { PropTypes } from 'ember-prop-types'
 import _ from 'lodash'
+import Ember from 'ember'
+const {Component, deprecate} = Ember
+import computed, {readOnly} from 'ember-computed-decorators'
+import PropTypeMixin, {PropTypes} from 'ember-prop-types'
+
+export default Component.extend(PropTypeMixin, {
```

Comment:
Are we _positive_ the mixin is necessary?  I haven't seen any adverse affects from just relying on the initializers